[Hannah] iP

kaixin-hc opened the pull request:
# Katheryne 
✨✨a genshin impact themed todolist tracker ✨✨
> Ad astra abyssosque! I am Katheryne, the receptionist here at the Adventurers' Guild.

Katheryne can help you remember your daily commissions, and even plan ahead. She can keep track of three kinds of tasks:
* Todos 
* Deadlines (With a deadline time to complete it by) 
* Events (With a date the event is at)

All you need to do is 
1. Download it from [here](https://github.com/kaixin-hc/ip)
2. Double click the jar file
3. **Get to adding tasks!**

## Features
- [x] Add, Delete, Read your tasks. Mark tasks as done.
- [x] Find Tasks by keywords
- [ ] Find Tasks by date added ( to come? )

## Commands
Here are a list of possible commands with Katheryne.
* Add tasks
  * Todos `todo <DESCRIPTION>`
  * Deadlines `deadline <DESCRIPTION> /by <DATE in YYYY-MM-DD format>`
  * Events `event <DESCRIPTION> /at <DATE in YYYY-MM-DD format>`
 * `list` -- List out current tasks
 * Actions on tasks
   * Find tasks by keyword `find <KEYWORD>`
   * Mark task as done by index `done <number>`
   * delete a task `done <number>`
 * `bye` Close the Katheryne Chatbot

Commit:
Add Gradle support

kaixin-hc committed:
feat: greet, echo, exit

solves level 1

kaixin-hc committed:
Feat: Implement add and list command

Solve Level 2.

I use ArrayList to store the list, as it is quite adaptable to changes
I might need to make in the future.

kaixin-hc committed:
Feat: Add ability to mark tasks as done

* Changed some words in the chatbot, so conversation flows more easily
* Created the Task class to store Task related methods

Might want to change the messages for markAsDone...

kaixin-hc committed:
Feat: Add task types

Each task type is a class which inherits from classes. These are todo,
deadline, and event; deadline has a by property which is when it is due
and event has an at property which is when the event is held.

These are created by starting the input with the keyword.

Also refactored the way the input was parsed.

Solves level-4 and A-inheritance

kaixin-hc committed:
Feat: Katheryne error handling

Solves level-5
Possible improvements:
* Can I make the error handling less wordy?
* Responses to incorrect deadline /error don't match Katheryne's regular tone
* Consider grouping UnknownCommandException under KatheryneExceptions

kaixin-hc committed:
Feat: Delete tasks + add some exceptions

kaixin-hc committed:
Tests, Errors: Automated UI testing + more exception handling

Originally used BufferedReader. Switched to Scanner because I was more
comfortable with it.

Added more descriptive Katheryne exceptions and made entering indexes
more comprehensive. Added automated UI testing for A-testing

kaixin-hc committed:
Chore: Remove unused import

kaixin-hc committed:
Merge branch 'master' into add-gradle-support

Test adding gradle on a branch that isn't master

kaixin-hc committed:
add jackson dependencies (for saving later)

kaixin-hc committed:
Feat: Save files as json & load existing task list

Before there was no way to save your list, which is not very
useful as the app would need to be running all the time.

Adding the ability to save and load is useful.

The jackson library is used to prepare me for the team project.

ngjiayuan commented on src/main/java/Katheryne.java:
```diff
@@ -0,0 +1,91 @@
+                    break;
+                } else if (keywordInput[0].equalsIgnoreCase("done") || keywordInput[0].equalsIgnoreCase("complete")) {
+                    if (keywordInput.length == 1) {
+                        throw new KatheryneExceptions("Traveller, please specify by index which task you completed.");
```
I think this line is too long, maybe you can consider wrapping the line?

ngjiayuan commented on src/main/java/Katheryne.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.*;
```
Should you list your imports explicitly for more clarity?

cyyeu replied:
+1 wildcard imports are not allowed according to Google's coding standard

ngjiayuan commented on src/main/java/Katheryne.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.*;
+
+public class Katheryne {
+    public static void main(String[] args) throws KatheryneExceptions {
```
Perhaps  split up the conditional statements to avoid long methods?

kaixin-hc replied:
Thanks for the suggestions! I've made the changes to imports but they're in a branch I haven't merged yet; I'll make sure to check the line wrapping as well before I next push.

Can I clarify what you mean by splitting up the conditional statements?

ngjiayuan reviewed (commented):
LGTM! Perhaps you could add javadoc to describe your methods?

cyyeu commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task{
```
Curly bracket should be preceded with a whitespace

cyyeu commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,13 @@
+public class Event extends Task{
```
Curly bracket should be preceded with a whitespace

cyyeu commented on src/main/java/Katheryne.java:
```diff
@@ -0,0 +1,91 @@
+                    }
+                } else if (userInput.equalsIgnoreCase("list")) {
+                    System.out.println("Here's the list I've stored for you:");
+                    for(int i = 1; i <= lst.size(); i++) {
```
'for' is not followed by whitespace

cyyeu commented on src/main/java/Katheryne.java:
```diff
@@ -0,0 +1,91 @@
+                    throw new UnknownCommandException();
+                }
+                System.out.println("There are currently " + lst.size() + " items in your list.");
+            } catch(UnknownCommandException e) {
```
'catch' is not followed by whitespace

cyyeu commented on src/main/java/Katheryne.java:
```diff
@@ -0,0 +1,91 @@
+                    String[] parsedDeadline = keywordInput[1].split(" /by ");
+                    if (parsedDeadline.length == 2) {
+                        lst.add(new Deadline(parsedDeadline[0], parsedDeadline[1]));
+                        System.out.println("'" + parsedDeadline[0] + "' added to your list, do by " + parsedDeadline[1]);
```
This line seems to be more than 120 characters and violates the style guide. Perhaps try to break up the text into two lines?

cyyeu reviewed (commented):
LGTM! Just needs some minor styling fixes. Perhaps you could run CheckStyle or use IntelliJ's "Reformat Code" feature to help you with that

kaixin-hc committed:
Read strings as dates

Gives more ability to extend date functionality in the future.

Still need to clean up code and include parsing for times, or using formats other than Y-M-D ...

kaixin-hc committed:
Merge branch 'branch-Level-7'

kaixin-hc committed:
Chore: Separate into packages

Solves A-Packages.

kaixin-hc committed:
Wip: Separate out Ui and Storage

kaixin-hc committed:
Wip: Add TaskList

kaixin-hc committed:
Add more OOP (Parser, Commands)

There are many different kinds of commands Katheryne can parse, which are differentiated
by a command word. The process to parse these commands are similar, with certain differences.

Separating these out into different classes, and putting all of the parsing code into it's own
class, increases the readability and the abstraction of the code. It also follows SLAP
principles.

This commit extracts out each command into it's own command under an abstract command class, and
modifies the main code to use this abstracted form. It also removes unused code blocks and further
groups methods which require communication with the user into the Ui class.

It also makes the commands case insensitive, for the user's convenience.

kaixin-hc committed:
Add Junit Tests for TaskList (WIP)

kaixin-hc committed:
Chore: Fix shadowjar config, add javadocs

kaixin-hc committed:
Docs: Add javadoc comments to many public methods

kaixin-hc committed:
Feat: Add the find command

When the user enters many tasks, it may be difficult to find the one you are looking for.

Using the find command allows users to find tasks more easily.

In the future, we might look at sorting tasks when they are returned, so that it is organised
more neatly and tasks can be even easier to find.

kaixin-hc committed:
Merge branch 'branch-Level-9' to add find feature

kaixin-hc committed:
Update README.md

kaixin-hc committed:
Implement checkstyle to help check coding standard.

chore: fix coding standard errors as well.

kaixin-hc committed:
Install JavaFX library

kaixin-hc committed:
Complete JavaFX tutorial (refactor to use .fxml files)

kaixin-hc committed:
Incorporate Katheryne logic

This commmit adds in the logic which was present in the CLI but not the GUI,
such as parsing the commands and printing out visually responses. It needs to
change for the GUI to be an appropriate substitute.

I have added the getResponse method to all commands, which returns a string
which will be handled by the GUI. I have also used Platform.exit() to close
the GUI during the bye command in handle user input.

I have updated some of the javadocs which are now deprecated as the methods
pertained to the CLI, which will no longer be actively edited.

There is still a lot to be improved, such as the look of the GUI as well as
the way that the application is closed. I also want to add a greeting message
when Katheryne is opened.

kaixin-hc committed:
Add greeting message

kaixin-hc committed:
Merge branch 'master' of github.com:kaixin-hc/ip to update the read.me

kaixin-hc committed:
Add assertions

kaixin-hc committed:
Shorten methods + Some SLAP

kaixin-hc committed:
Merge pull request #2 from kaixin-hc/branch-A-CodeQuality

Shorten methods + Some SLAP

kaixin-hc committed:
Merge pull request #3 from kaixin-hc/branch-A-Assertions

Add assertions

kaixin-hc committed:
detect and prevent duplicates

kaixin-hc committed:
Chore: Update javadocs

kaixin-hc committed:
Better GUI

kaixin-hc committed:
Add User Guide & Ui image

kaixin-hc committed:
Update Ui
